### PR TITLE
[#18] Create demo user functionality

### PR DESCRIPTION
**Pull request by denjj**
### What this PR does (required):
- Creates a guest login button which allows signing in as a hardcoded user for demo purposes
- Implements the button on login and signup pages
- The button will have a circular progress bar when submitting

### Screenshots / Videos (front-end only):
image
image

### Any information needed to test this feature (required):
- Fill out the hardcoded login credentials in './client/src/pages/Login/GuestLogin/GuestLogin.tsx' with a valid and existing email/password
- In the browser, click the 'Continue as guest' button on either the login or signup pages

### CODE REVIEW

**Comment by ezeilo-su** [comment]
Please link your ticket

File: client/src/pages/Login/GuestLogin/GuestLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { useAuth } from '../../../context/useAuthContext';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import useStyles from './useStyles';
+
+export default function LoginGuest(): JSX.Element {
+  const { updateLoginContext } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+  const classes = useStyles();
+  const guestLoginCredentials = {
+    // login credentials to be hardcoded
```
Comment (ezeilo-su) [inline client/src/pages/Login/GuestLogin/GuestLogin.tsx:15]: We don't need this comment. Please remove it :smiley:

**Review by ezeilo-su — COMMENTED** [review]
UI looks good!

File: client/src/pages/Login/GuestLogin/GuestLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import login from '../../../helpers/APICalls/login';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import CircularProgress from '@mui/material/CircularProgress';
+import { useAuth } from '../../../context/useAuthContext';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import useStyles from './useStyles';
+
+export default function LoginGuest(): JSX.Element {
```
Comment (rajivtitus) [inline client/src/pages/Login/GuestLogin/GuestLogin.tsx:10]: Excellent work componentizing this 💯

**Review by rajivtitus — APPROVED** [review]
Great work! 
Good use of MUI and React too 👍🏽

**Comment by rajivtitus** [comment]
Hey @denjj , let's merge this PR in when you get a chance 👍🏽

**Pull request merged**